### PR TITLE
Adds support for Snowflake databases

### PR DESCRIPTION
**Pull request by edgararuiz**
- Adds a formal Snowflake "back-end" script 
- Customizes `log10()` translation 
- Adds simulate function
- Prevent ANALYZE TABLE to run as part of `copy_to()` by creating a method with an empty output
- Adds single test for `log10()` SQL output

### CODE REVIEW

**Comment by edgararuiz** [comment]
@hadley PTAL

File: R/backend-snowflake.R [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+      base_win
+  )
+}
+
+#' @export
+#' @rdname backend-snowflake
+simulate_snowflake <- function() simulate_dbi("Snowflake")
+
+#' @export
+sql_table_analyze.Snowflake <- function(con, table, ...) {}
```
Comment (hadley) [inline R/backend-snowflake.R:36]: Could you add a comment here explaining why, preferably with a link to snowflake docs?

Comment (edgararuiz) [reply R/backend-snowflake.R:39]: This was like trying to prove a negative.  Could not find an explicit acknowledgement in the docs that we don't need to run the table analysis.  I commented and included the link to *all* SQL commands, hopefully that helps.

**Review by hadley — APPROVED** [review]
Can you please add a bullet to the top of `NEWS.md`? It should briefly [describe the change](http://style.tidyverse.org/news.html) and end with `(@yourname, #issuenumber)`.

**Pull request merged**